twilight-mention/src/parse/impl.rs

```diff
@@ -201,6 +329,52 @@ impl ParseMention for Id<UserMarker> {
     }
 }
 
+/// Matches the four segments of [`CommandMention::parse`] into the final `Result` for it.
+#[inline]
+fn match_command_mention_from_segments<'s>(
+    id: Id<CommandMarker>,
+    first: Option<&'s str>,
+    second: Option<&'s str>,
+    third: Option<&'s str>,
+    fourth: Option<&'s str>,
+) -> Result<CommandMention, ParseMentionError<'s>> {
+    match (first, second, third, fourth) {
+        (_, _, _, Some(extra)) => Err(ParseMentionError {
+            kind: ParseMentionErrorType::ExtraneousPart { found: extra },
+            source: None,
+        }),
+        (None, _, _, _) => {
+            Err(ParseMentionError {
+                kind: ParseMentionErrorType::PartMissing {
+                    // at least two for command
+                    expected: 2,
+                    // we found the id until now
+                    found: 1,
+                },
+                source: None,
+            })
+        }
+        (Some(name), None, None, None) => Ok(CommandMention::Command {
+            name: name.to_string(),
```

Comment:
we should use `.to_owned` here, as thats what twilight generally uses

Reply:
done